engine/common/follower/pending_tree/pending_tree.go

```diff
@@ -81,20 +95,22 @@ func NewPendingTree(finalized *flow.Header) *PendingTree {
 	}
 }
 
-// AddBlocks accepts a batch of certified blocks, adds them to the tree of pending blocks and finds blocks connected to the finalized state.
-// This function performs processing of incoming certified blocks, implementation is split into a few different sections
-// but tries to be optimal in terms of performance to avoid doing extra work as much as possible.
-// This function proceeds as follows:
-//  1. Sorts incoming batch by height. Since blocks can be submitted in random order we need to find blocks with
-//     the lowest height since they are candidates for being connected to the finalized state.
-//  2. Filters out blocks that are already finalized.
-//  3. Deduplicates incoming blocks. We don't store additional vertices in tree if we have that block already stored.
-//  4. Checks for exceeding byzantine threshold. Only one certified block per view is allowed.
-//  5. Finally, blocks with the lowest height from incoming batch that connect to the finalized state we will
-//     mark all descendants as connected, collect them and return as result of invocation.
+// AddBlocks accepts a batch of certified blocks, adds them to the tree of pending blocks and finds blocks connected to
+// the finalized state.
+//
+// Details:
+// Adding blocks might result in additional blocks now being connected to the latest finalized block. The returned
+// slice contains:
+//  1. the subset of `certifiedBlocks` that are connect to the finalized block
```

Comment:
```suggestion
//  1. the subset of `certifiedBlocks` that are newly connected to the finalized block
```

Reply:
@jordanschalm : to be 100% technically accurate, do you think it is really only the _newly_ connected blocks? 

If you call `AddBlock` repeatedly with duplicated blocks, the blocks might already be connected. Though, in a later call, duplicated blocks would still be returned as part of the  `certifiedBlocks` list as they continue to pass this check: https://github.com/onflow/flow-go/blob/08aad74c26df77a99f5237f0258b11f19252c6bd/engine/common/follower/pending_tree/pending_tree.go#L151.
Therefore, I think that the following would be the most technically accurate: 
```
the subset of `certifiedBlocks` that are newly connected to the finalized block
```

what do you think?

Reply:
added clarifying comment in https://github.com/onflow/flow-go/pull/4114/commits/3c3f3deca28e8095d186012e23fb0293b03c4df5